### PR TITLE
Editor: Standardize header button size to 32px

### PR DESCRIPTION
**Pull request by t-hamano**
Part of #58293
Related to #56635

## What?

This PR standardizes the button size to 32px in the header of Post Editor, Site Editor, and Widget Editor.

## Why?

As part of the unification of UI component sizes proposed in #46734

## Testing Instructions

Previous behavior should not be affected. Check that there are no buttons other than 32px in all editors.

## Screenshots or screencast 

The biggest change due to this PR is the Widget Editor. In the screenshot below, the focus style is applied to all buttons in the header and the size changes are compared.

### Before

image

### After

image

### CODE REVIEW

File: packages/edit-widgets/src/components/header/style.scss [hunk 1]
```diff
@@ -103,15 +103,18 @@
...
 	// here to the original button styles
 	// Specificity bump needed to offset https://github.com/WordPress/gutenberg/blob/8ea29cb04412c80c9adf7c1db0e816d6a0ac1232/packages/components/src/toolbar/style.scss#L76
 	> .components-button.has-icon.has-icon.has-icon,
 	> .components-dropdown > .components-button.has-icon.has-icon {
-		height: $button-size;
-		min-width: $button-size;
-		padding: 6px;
+		height: $button-size-compact;
+		min-width: $button-size-compact;
+		padding: 4px;
```
Comment (t-hamano) [inline packages/edit-widgets/src/components/header/style.scss:117]: This padding change follows this approach:

https://github.com/WordPress/gutenberg/blob/b5a31264e7c614534eb021ca272e9dca30676b70/packages/editor/src/components/document-tools/style.scss#L47

File: packages/edit-widgets/src/components/header/style.scss [hunk 2]
```diff
@@ -103,15 +103,18 @@
 
 .edit-widgets-header-toolbar {
 	gap: $grid-unit-10;
+	// Some plugins add buttons here despite best practices.
+	// Push them a bit rightwards to fit the top toolbar.
+	margin-right: $grid-unit-10;
```
Comment (t-hamano) [inline packages/edit-widgets/src/components/header/style.scss:108]: This change follows this approach:

https://github.com/WordPress/gutenberg/blob/b5a31264e7c614534eb021ca272e9dca30676b70/packages/editor/src/components/document-tools/style.scss#L70-L72

This margin also prevents the focus from being cut off in the Widget Editor.

image

**Comment by afercia** [comment]
The 'Preview' button (actually a link) is shown on small screens and it's still 36 pixels:

Screenshot 2024-02-01 at 09 44 57

From an user perspective, it's really not clear what the difference is between all the 'View', 'Preview', 'View Post' buttons but that's a separate issue...

**Comment by t-hamano** [comment]
> The 'Preview' button (actually a link) is shown on small screens and it's still 36 pixels:

Thank you for finding it! Fixed with [dd75a3d](https://github.com/WordPress/gutenberg/pull/58532/commits/dd75a3d44ba14cf0195bea5b2c9e6958c2ade30c).

**Review by noisysocks — APPROVED** [review]
Looks good to me 👍 I tested the post editor, site editor, and widgets editor.

Would be good to get @jasmussen to check that this is all going according to his plan 🙂

**Comment by jasmussen** [comment]
Ship it as fast as you can! And thank you!

**Pull request merged**